### PR TITLE
Trigger CI when workflow files are changed

### PR DESCRIPTION
**Pull request by maxhoerstr**
Changes of the workflow files now triggers the execution of the whole pipeline

### CODE REVIEW

File: .github/workflows/build_and_deploy_docu.yml [hunk 1]
```diff
@@ -2,10 +2,9 @@ name: Build and deploy docu
 
 on:
   push:
-    branches:
-      - 'main'
```
Comment (marbre) [inline .github/workflows/build_and_deploy_docu.yml:6]: Do you really intend to build and deploy also for branches other than main?

File: .github/workflows/code_coverage.yml [hunk 2]
```diff
@@ -4,11 +4,13 @@ on:
   push:
     paths:
       - 'tests/**'
+      - '.github/workflows/**'
```
Comment (marbre) [inline .github/workflows/code_coverage.yml:7]: What about changes to the source code? I assume these could trigger the code coverage action as well, even though not part of issue #35.

File: .github/workflows/code_coverage.yml [hunk 3]
```diff
@@ -4,11 +4,13 @@ on:
...
   pull_request:
     branches:
       - 'main'
     paths:
       - 'tests/**'
+      - '.github/workflows/**'
```
Comment (marbre) [inline .github/workflows/code_coverage.yml:13]: Same as above.

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -1,16 +1,16 @@
 name: Lint
 
-on:
+on:
```
Comment (marbre) [inline .github/workflows/lint.yml:3]:
```suggestion
on:
```
White space error at EOL.

**Review by marbre — CHANGES_REQUESTED** [review]

File: .github/workflows/build_and_deploy_docu.yml (hunk 1 above)
Comment (maxhoerstr) [reply .github/workflows/build_and_deploy_docu.yml:6]: I thought about that too, but otherwise, the CI will not be triggered, right?

Comment (marbre) [reply .github/workflows/build_and_deploy_docu.yml:6]: Why not just giving it a try? You'll see what happens to your PR.

File: .github/workflows/code_coverage.yml (hunk 2 above)
Comment (maxhoerstr) [reply .github/workflows/code_coverage.yml:7]: That should of course trigger the action! You are right, this should be handled in another issue / branch.

File: .github/workflows/code_coverage.yml (hunk 3 above)
Comment (maxhoerstr) [reply .github/workflows/code_coverage.yml:13]: +1

File: .github/workflows/build_and_deploy_docu.yml (hunk 1 above)
Comment (marbre) [reply .github/workflows/build_and_deploy_docu.yml:6]: Anyway, do you want a docu push on *every* PR?

Comment (maxhoerstr) [reply .github/workflows/build_and_deploy_docu.yml:6]: I will undo the changes and apply your recommendation in another issue / branch.

**Review by marbre — APPROVED** [review]
LGTM! Please make sure to squash when merging.

**Pull request merged**